src/main/Core/Dock/DockModule.ts

```diff
@@ -1,7 +1,24 @@
-import type { App } from "electron";
+import type { Dependencies } from "@Core/Dependencies";
+import type { DependencyRegistry } from "@Core/DependencyRegistry";
 
 export class DockModule {
-    public static bootstrap(app: App) {
-        app.dock?.hide();
+    public static bootstrap(dependencyRegistry: DependencyRegistry<Dependencies>) {
+        const app = dependencyRegistry.get("App");
+        const settingsManager = dependencyRegistry.get("SettingsManager");
+        const showAppIconInDock = settingsManager.getValue("appearance.showAppIconInDock", false);
+        this.toggleAppIconInDock(app, showAppIconInDock);
+
+        const eventSubscriber = dependencyRegistry.get("EventSubscriber");
+        eventSubscriber.subscribe("settingUpdated[appearance.showAppIconInDock]", ({ value }: { value: boolean }) => {
+            this.toggleAppIconInDock(app, value);
+        });
+    }
+
+    private static toggleAppIconInDock(app: Electron.App, value: boolean) {
```

Comment:
Please use a variable name that is more descriptive.

```suggestion
    private static toggleAppIconInDock(app: Electron.App, showAppIconInDock: boolean) {
```